### PR TITLE
[Swoole PSR7] Fix ServerRequest header Nyholm PSR7

### PR DESCRIPTION
**Pull request by pistej**
When using swoole app with PSR7, heades are not correctly set from Swoole\Http\Request to \Nyholm\Psr7\ServerRequest

### CODE REVIEW

**Comment by pistej** [comment]
Hi @Nyholm, can you check this PR ? thx

File: src/swoole-nyholm/src/RequestHandlerRunner.php [hunk 1]
```diff
@@ -39,7 +39,7 @@ public function handle(Request $request, Response $response): void
         $psrRequest = (new \Nyholm\Psr7\ServerRequest(
             $request->getMethod(),
             $request->server['request_uri'] ?? '/',
-            array_change_key_case($request->server ?? [], CASE_UPPER),
+            $request->header ?? [],
```
Comment (alexander-schranz) [inline src/swoole-nyholm/src/RequestHandlerRunner.php:42]: The `CASE_UPPER` looks strange here. If I look at the the ServerRequest, it always converts uppercase to lowercase so `CASE_LOWER` would maybe make more sense: https://github.com/Nyholm/psr7/blob/f734364e38a876a23be4d906a2a089e1315be18a/src/MessageTrait.php#L147

It is also converted in other class to uppercase here: https://github.com/php-runtime/runtime/blob/d2a9d3e2e5d13d6aa63e0edc97e0c2df175b3916/src/swoole/src/SymfonyHttpBridge.php#L30

@Nyholm what do you think?

If we change it we would need todo this in all classes to be consistent.

@pistej What type of error you are trying to solve. Can you try to create a test case? Without a test case things get lost over the time.

Comment (pistej) [reply src/swoole-nyholm/src/RequestHandlerRunner.php:42]: the error is, that headers from any request send to swoole server are not available in Nyholm ServerRequest -> here is wrong assignmen, dont know why `$request->server` is set to  $header parameter

Comment (alexander-schranz) [reply src/swoole-nyholm/src/RequestHandlerRunner.php:42]: I see, why remov ethe `array_change_key_case`? can we not keep that then?

Comment (alexander-schranz) [reply src/swoole-nyholm/src/RequestHandlerRunner.php:42]: Still the same here: https://github.com/php-runtime/runtime/blob/d2a9d3e2e5d13d6aa63e0edc97e0c2df175b3916/src/swoole/src/SymfonyHttpBridge.php#L30

Is there any difference between swoole and openswoole here?

Comment (pistej) [reply src/swoole-nyholm/src/RequestHandlerRunner.php:42]: think headers keys are case-insensitive besed on RFC 7230 and more validation is done in ServerRequest `__construct`, `setHeader` is called and then `validateAndTrimHeader` from MessageTrait, so just assign without modifying is ok
https://github.com/Nyholm/psr7/blob/master/src/MessageTrait.php#L138

Comment (pistej) [reply src/swoole-nyholm/src/RequestHandlerRunner.php:42]: in SymfonyHttpBridge there is `new SymfonyRequest` and on this line 30 is set `$server` in construct, this assingnemt is ok, server to server, and ServeBag is created https://github.com/symfony/http-foundation/blob/6.1/Request.php#L274 where all upsercasse keys are used, which is good https://github.com/symfony/http-foundation/blob/6.1/ServerBag.php#L26

Comment (pistej) [reply src/swoole-nyholm/src/RequestHandlerRunner.php:42]: ano no, there's no diff. between swoole and openswoole in this case

**Review by Nyholm — APPROVED** [review]
Thank you

**Pull request merged**